watertap/unit_models/tests/test_cstr_injection.py

```diff
@@ -467,3 +482,642 @@ def test_error_without_oxygen():
+        ] == pytest.approx(1e2, rel=1e-8)
+        assert sfx_rxn[
+            model.fs.unit.control_volume.reactions[0.0].rate_expression["R8"]
+        ] == pytest.approx(1e2, rel=1e-8)
+
+        # Check that unit model has scaling factors
+        sfx_cv = model.fs.unit.control_volume.scaling_factor
+        assert isinstance(sfx_cv, Suffix)
+        assert len(sfx_cv) == 1
+        assert sfx_cv[model.fs.unit.control_volume.volume[0]] == pytest.approx(
+            1e-3, rel=1e-3
+        )
+
+        sfx_unit = model.fs.unit.scaling_factor
+        assert isinstance(sfx_unit, Suffix)
+        assert len(sfx_unit) == 3
+        assert sfx_unit[model.fs.unit.eq_hydraulic_retention_time[0]] == pytest.approx(
+            0.00047796296, rel=1e-8
+        )
+        assert sfx_unit[model.fs.unit.eq_mass_transfer[0]] == pytest.approx(
+            0.004, rel=1e-8
+        )
+        assert sfx_unit[model.fs.unit.eq_electricity_consumption[0]] == pytest.approx(
+            0.09, rel=1e-8
+        )
+
+    # TODO: Remove test once iscale is deprecated
+    @pytest.mark.integration
+    def test_example_case_iscale(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+
+        m.fs.properties = ASM1ParameterBlock()
+        m.fs.reactions = ASM1ReactionParameterBlock(property_package=m.fs.properties)
+
+        m.fs.unit = CSTR_Injection(
+            property_package=m.fs.properties,
+            reaction_package=m.fs.reactions,
+            has_aeration=True,
+            electricity_consumption=ElectricityConsumption.calculated,
+        )
+
+        m.fs.unit.inlet.flow_vol.fix(20648 * units.m**3 / units.day)
+        m.fs.unit.inlet.temperature.fix(308.15 * units.K)
+        m.fs.unit.inlet.pressure.fix(1 * units.atm)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_I"].fix(27 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_S"].fix(58 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_I"].fix(92 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_S"].fix(363 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_BH"].fix(50 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_BA"].fix(0 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_P"].fix(0 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_O"].fix(0 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_NO"].fix(0 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_NH"].fix(23 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_ND"].fix(5 * units.g / units.m**3)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_ND"].fix(16 * units.g / units.m**3)
+        m.fs.unit.inlet.alkalinity.fix(7 * units.mol / units.m**3)
+
+        m.fs.unit.volume.fix(500)
+        m.fs.unit.injection.fix(0)
+        m.fs.unit.injection[0, "Liq", "S_O"].fix(2e-3)
```

Comment:
perturbing injection rate of oxygen would be useful when using scaling profiler

Reply:
Addressed